ofrak_core/ofrak/core/gzip.py

```diff
@@ -41,44 +39,51 @@ class GzipUnpacker(Unpacker[None]):
 
     async def unpack(self, resource: Resource, config=None):
         data = await resource.get_data()
-        # GzipFile is faster (spawning external processes has overhead),
-        # but pigz is more willing to tolerate things like extra junk at the end
-        try:
-            with GzipFile(fileobj=BytesIO(data), mode="r") as gzip_file:
-                return await resource.create_child(
-                    tags=(GenericBinary,),
-                    data=gzip_file.read(),
-                )
-        except BadGzipFile:
-            # Create temporary file with .gz extension
-            with tempfile.NamedTemporaryFile(suffix=".gz") as temp_file:
-                temp_file.write(data)
-                temp_file.flush()
-                cmd = [
-                    "pigz",
-                    "-d",
-                    "-c",
-                    temp_file.name,
-                ]
-                proc = await asyncio.create_subprocess_exec(
-                    *cmd,
-                    stdout=asyncio.subprocess.PIPE,
-                    stderr=asyncio.subprocess.PIPE,
-                )
-                stdout, stderr = await proc.communicate()
-                data = stdout
-                if proc.returncode:
-                    # Forward any gzip warning message and continue
-                    if proc.returncode == -2 or proc.returncode == 2:
-                        LOGGER.warning(stderr)
-                        data = stdout
-                    else:
-                        raise CalledProcessError(returncode=proc.returncode, cmd=cmd, stderr=stderr)
-
-                await resource.create_child(
-                    tags=(GenericBinary,),
-                    data=data,
-                )
+        pigz_installed = await PIGZ.is_tool_installed()
+        if len(data) >= 1024 * 1024 * 4 and pigz_installed:
+            uncompressed_data = await self.unpack_with_pigz(data)
+        else:
+            try:
+                uncompressed_data = await self.unpack_with_zlib_module(data)
+            except Exception:  # pragma: no cover
+                if not pigz_installed:
+                    raise
+                uncompressed_data = await self.unpack_with_pigz(data)
```

Comment:
Since `unpack_with_zlib_module` now handles the trailing bytes, I'm not following why we are catching exceptions here?

It seems like this logic could be simplified to:

```suggestion
        pigz_installed = await PIGZ.is_tool_installed()
        if pigz_installed and len(data) >= 1024 * 1024 * 4:
            uncompressed_data = await self.unpack_with_pigz(data)
        else:
            uncompressed_data = await self.unpack_with_zlib_module(data)
```

Reply:
I had that originally, but I figured in case of some other format incompatibility or big in python we're not aware of might as well have the fallback still.

Reply:
no longer using PIGZ for decompression